lib/Catmandu/Store/Datahub/Bag.pm

```diff
@@ -70,7 +73,11 @@ sub get {
 # Create a new record
 sub add {
     my ($self, $data) = @_;
-    return $self->api->update($data->{'id'}, $data->{'_'});
+
+    if ($self->store->raw_parser == 1) {
+      return $self->api->add($data->{'_metadata'});
```

Comment:
I don't think the Datahub API should make any assumptions about the datamodel of the incoming data. the _metadata construct is a typical ouptput of OAI. I think this would violate the[ open/closed principle](https://en.wikipedia.org/wiki/Open/closed_principle).

Maybe a separate optional parser object could be injected here. Maybe not in the API implementation itself, but right before the data is passed of to this method.

Reply:
Agreed, should be part of the transformation (perhaps in a fix?).